### ISSUE TITLE
Necessity of resetting the MCU?

### ISSUE THREAD

**Issue by Ravenstine** [TronixLab/SFM3300#1]
Firstly, thanks for writing this library!

One thing I noticed, due to what may be a flaw in my circuit, is that the library resets the microcontroller unit when the sensor reading fails:

https://github.com/TronixLab/SFM3300/blob/main/SFM3300.cpp#L90-L96

Maybe consider removing this?  It threw me for a loop (figuratively and literally!) until I read through the code.  I'm not sure if it's a typical use case, but I think the reset makes it impossible to hot-swap flow sensors without resetting runtime state.

**Comment by TronixLab** [comment]
Hi Ravenstine,

Thanks for your review, the library I made was tested many times (continuously for few hours), and it work reliable. Soft resetting the sensor in not enough to ensure the reliability, it required hard resetting (reset the MCU). You may refer to product specification. 
https://github.com/TronixLab/SFM3300/blob/main/docs/Sensirion_Mass_Flow_Meters_SFM3300_Datasheet.pdf

**Comment by TronixLab** [comment]
Yes your right it is not possible to hot-swap the flow sensor without resetting the MCU. Soft-reset does not work alone.
